### PR TITLE
Limit amount of concurrent requests

### PR DESCRIPTION
**Pull request by BrennanConroy**
@davidfowl @Tratcher @danroth27 

Goes on top of #2562

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs [hunk 1]
```diff
@@ -25,6 +26,10 @@ internal class HttpSource
         private readonly string _password;
         private readonly Reports _reports;
 
+        // Limiting concurrent requests to limit the amount of files open at a time on Mac OSX
+        // the default is 256 which is easy to hit if we don't limit concurrency
+        private static SemaphoreSlim _throttle = new SemaphoreSlim(50);
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: Makr 50 a constant.

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs [hunk 2]
```diff
@@ -25,6 +26,11 @@ internal class HttpSource
         private readonly string _password;
         private readonly Reports _reports;
 
+        private const int _concurrencyLimit = 50;
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: No _ for constants

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs [hunk 3]
```diff
@@ -25,6 +26,11 @@ internal class HttpSource
         private readonly string _password;
         private readonly Reports _reports;
 
+        private const int concurrencyLimit = 50;
```
Comment (pranavkm) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: `ConcurrencyLimit`

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs [hunk 4]
```diff
@@ -25,6 +26,11 @@ internal class HttpSource
...
+        // Limiting concurrent requests to limit the amount of files open at a time on Mac OSX
+        // the default is 256 which is easy to hit if we don't limit concurrency
+        private static SemaphoreSlim _throttle = new SemaphoreSlim(concurrencyLimit);
```
Comment (pranavkm) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: `readonly`

**Comment by pranavkm** [comment]
:shipit:

**Comment by BrennanConroy** [comment]
f3103122c058632007ce915a2edd0884130d427b

**Pull request closed without merging**